robot/impl/local_robot.go

```diff
@@ -1174,15 +1175,53 @@ func (r *localRobot) applyLocalModuleVersions(cfg *config.Config) {
 }
 
 func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config, forceSync bool) {
+	var allErrs error
+
+	// Maintenance config can be configured to block reconfigure based off of a sensor reading
+	// These sensors can be configured on the main robot, or a remote
+	// In situations where there are conflicting sensor names the following behavior happens
+	// Main robot and remote share sensor name -> main robot sensor is chosen
+	// Only remote has the sensor name -> remote sensor is read
+	// Multiple remotes share a senor name -> conflict error is returned and reconfigure happens
+	// To specify a specific remote sensor use the name format remoteName:sensorName to specify a remote sensor
+	if newConfig.MaintenanceConfig != nil {
+		name, err := resource.NewFromString(newConfig.MaintenanceConfig.SensorName)
+		if err != nil {
+			r.logger.Warnf("Sensor Name %s is not in a supported format", newConfig.MaintenanceConfig.SensorName)
+		} else {
+			sensorComponent, err := robot.ResourceFromRobot[sensor.Sensor](r, name)
+			if err != nil {
+				r.logger.Warnf("%s, Starting reconfiguration", err.Error())
+			} else {
+				canReconfigure, err := r.checkMaintenanceSensorReadings(ctx, newConfig.MaintenanceConfig.MaintenanceAllowedKey, sensorComponent)
+				if !canReconfigure {
+					r.logger.Info("maintenanceAllowedKey found from readings on maintenance sensor. Reconfigure disabled")
```

Comment:
all comments/logs with maintenanceAllowedKey should be updated to `maintenance_allowed_key`

Reply:
```suggestion
					r.logger.Info("maintenanceAllowedKey found from readings on maintenance sensor. Skipping reconfiguration.")
```
logs should be clear that skipping or continuing with reconfiguration is a one time thing and will be reevaluated next time.